### PR TITLE
Bugfixes for error handling towards Customer API

### PR DESCRIPTION
**Pull request by norrs**
This needs to bubble up an exception so we return non successfull http
status code to Cognito.

This could probably require another refactor to make it more nice.....

### CODE REVIEW

**Comment by norrs** [comment]
Not nicest work... but we want this out to actually fail cognito error. 

There is another PR coming up after this which actually fixes the current bug where customerId and cristinId didnt get populated for valid customers.

**Comment by norrs** [comment]
This whole API client should in the future come from a generated openapi client, so we dogfood openapi.

**Comment by norrs** [comment]
I will make a task on the board for follow up with a refactor, abusing my administrator rights to get going. @janbj @brinxmat @axthosarouris @svenngjeto

File: cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/CustomerApiClient.java [hunk 1]
```diff
@@ -48,15 +43,27 @@ public CustomerApiClient(HttpClient httpClient,
     @Override
     public Optional<CustomerResponse> getCustomer(String orgNumber) {
         logger.info("Requesting customer information for orgNumber: " + orgNumber);
-        return fetchCustomerInformation(orgNumber)
-            .stream()
-            .filter(responseIsSuccessful())
-            .map(tryParsingCustomer())
-            .findAny()
-            .flatMap(this::getValueOrLogError);
+        try {
```
Comment (axthosarouris) [inline cognito-pre-token-generation/src/main/java/no/unit/nva/cognito/service/CustomerApiClient.java:46]: Hi. It looks OK, except for the error messages that could be constants.

However, if you want to simplify the handling of optionals and stuff you can refactor a bit. 

```
 @Override
    public Optional<CustomerResponse> getCustomer(String orgNumber) {
        logger.info("Requesting customer information for orgNumber: " + orgNumber);

        var response = fetchCustomerInformation(orgNumber);
        checkFoValidResponse(response);
        return Optional.ofNullable(parseCustomer(response));
    }

    private void checkFoValidResponse(HttpResponse<String> response) {
        if (responseIsUnexpected(response)) {
            logger.error("Error fetching customer information, API response was {}", response.statusCode());
            throw new IllegalStateException("Error fetching customer information");
        }
    }

    private boolean responseIsUnexpected(HttpResponse<String> response) {
        return response.statusCode() != HttpStatus.SC_OK && response.statusCode() != HttpStatus.SC_NOT_FOUND;
    }

    private IllegalStateException getHttpClientInitializationError() {
        logger.error("Error fetching customer information");
        return new IllegalStateException("Error fetching customer information, http client failed to initialize.");
    }

    private HttpResponse<String> fetchCustomerInformation(String orgNumber) {
        return Try.of(formUri(orgNumber))
            .map(URIBuilder::build)
            .map(this::buildHttpRequest)
            .map(this::sendHttpRequest)
            .orElseThrow(fail -> getHttpClientInitializationError());
    }

    private CustomerResponse parseCustomer(HttpResponse<String> response) {
        if (response.statusCode() == HttpStatus.SC_NOT_FOUND) {
            return null;
        }
        return attempt(() -> objectMapper.readValue(response.body(), CustomerResponse.class))
            .orElseThrow(fail -> handleJsonProcessingError(fail.getException()));
    }

    private IllegalStateException handleJsonProcessingError(Exception e) {
        logger.error("Error parsing customer information", e);
        return new IllegalStateException("Error parsing customer information");
    }
```

**Review by axthosarouris — APPROVED** [review]
i do not particularly like the copy-pasting of the error strings but OK.  I have submitted a quick suggestion for making the code a bit clearer, bu it is optional

**Pull request merged**